src/Http/Http.Abstractions/src/RouteHandlerFilterContext.cs

```diff
@@ -0,0 +1,35 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Microsoft.AspNetCore.Http;
+
+/// <summary>
+/// Provides an abstraction for wrapping the <see cref="HttpContext"/> and parameters
+/// provided to a route handler.
+/// </summary>
+public class RouteHandlerFilterContext
```

Comment:
```suggestion
public sealed class RouteHandlerFilterContext
```

or readonly struct?

Reply:
We anticipate that we will probably be inheriting from this class to produce the generic variant of the `Parameters` object so I don't think we can make it `sealed`.